File: Easy/Reverse Bits/reverse-bits.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            Long X = Long.parseLong(read.readLine());

            Solution ob = new Solution();
            System.out.println(ob.reversedBits(X));
        }
    }
}
// } Driver Code Ends


// User function Template for Java

class Solution {
    static Long reversedBits(Long n) {
        // code here
        long s = 0;
        for(int i = 1 ; i <= 32 ; i++)
        {
            s <<= 1;
            s |= (n&1);
            n >>= 1;
        }
        return s;
    }
};

File: Difficulty: Easy/Reverse Bits/reverse-bits.py
#User function Template for python3

class Solution:
    def reverseBits(self,n):
        #code here
        s = bin(n)
        s = s[2:]
        s = s[::-1]
        
        res = int(s, 2)
        return res
        


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math

def main():
    
    T=int(input())
    
    while(T>0):
        
        
        n=int(input())
        
        ob=Solution()
        print(ob.reverseBits(n))
     
        T-=1

if __name__=="__main__":
    main()
# } Driver Code Ends